### DISCUSSION TITLE
Question about torch_geometric.transforms.RadiusGraph

### DISCUSSION THREAD

**Discussion by wwyi1828** [pyg-team/pytorch_geometric#4494] [Q&A] [answered]
Hi, 
I created a sample graph with only 9 nodes as below. In my expectation, 24 edges should be created when the radius=1. However, 0 edges will be created when radius < sqrt(2). When radius = sqrt(2), the 24 edges will be created, but more edges are expected to be created. 

```
from torch_geometric.data import Data as gData
import torch_geometric.transforms as T
pre_trans = T.RadiusGraph(1)
test_graph = gData(x=torch.tensor(np.array([0,1,2,3,4,5,6,7,8])),
                   pos=torch.tensor(np.array([[0,0],[0,1],[0,2],[1,0],[1,1],[1,2],[2,0],[2,1],[2,2]])))
pre_trans(test_graph)
test_graph.edge_index
```
could you tell me how the radius works in `RadiusGraph` and other related functions/classes? Thanks!

**Comment by rusty1s** [comment, marked as answer]
It looks like you need to apply the following fixes:
* `data.pos` needs to be of type floating-point
* `Radius(1)` will only include connections with distance less than `1`. You can add a small value to fix this
```python
import math

import numpy as np
import torch

import torch_geometric.transforms as T
from torch_geometric.data import Data as gData

pre_trans = T.RadiusGraph(math.sqrt(2) + 1e-5)
test_graph = gData(
    x=torch.tensor(np.array([0, 1, 2, 3, 4, 5, 6, 7, 8])), pos=torch.tensor(
        np.array([[0, 0], [0, 1], [0, 2], [1, 0], [1, 1], [1, 2], [2, 0],
                  [2, 1], [2, 2]])))
test_graph.pos = test_graph.pos.to(torch.float)
pre_trans(test_graph)
print(test_graph.edge_index.shape)
```

**Reply by wwyi1828** [reply]
Thanks for the reply! That works
